### PR TITLE
Heartrate deamon

### PR DESCRIPTION
**Pull request by michaelcypher**
Instead of starting a new process that takes minutes to load a model, a deamon constantly runs in the container. The server starts a cheap subprocess that communicates with the deamon using a ZeroMQ message queue which requests a video and gets streaming the raw frame data back. This gets pipe by the server to the ffmpeg process. At the moment this is streaming frames between 15-20 fps which is not very good but there is not the 2 minute wait.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -12,13 +11,16 @@ def index():
 
 @app.route('/video')
 def video():
+    print "Video request"
+
     FNULL = open(os.devnull, 'w')
 
-    hri_cmd = ['python', 'heart_rate_deamon.py', video_file, opencv_path]
-    opencv_ps = subprocess.Popen(hri_cmd, stdout=subprocess.PIPE, stderr=FNULL)
+    client_proc_cmd = 'python heart_rate_client.py ' + video_file
```
Comment (kren1) [inline app.py:18]: I'm a bit confused as to what is the purpose of this? Why have python_heart_rate_clieant.py a subproccess where you could just have it as a function.

Comment (michaelcypher) [inline app.py:18]: Thought you would ask. Basically, we need to write the output of the message queue to the ffmpeg process. I tried doing this with a generator but it expects a pipe (file) so I couldn't. 

There were 2 ways I thought of doing this: 

1) have this in a function that is run in another thread and is constantly writing to the ffmpeg process or 

2) have it in another process where I can just hook up the processes output into ffmpeg input. 

I didn't try 1 but I suspected there would be issues with the global interpreter lock that Python has (which means any I/O doesn't benefit from mulithreading I believe). I was thinking I would try this approach once I got streaming from YouTube to work and see if I sped up the output streaming.

Comment (kren1) [inline app.py:18]: Well I guess the server could directly write the output to the ffmpeg, but that breaks the queue idea. Or maybe the os could be convinced to treat ZeroMQ socket as a pipe. I'm not 100% happy with this now, but it is also not an important issue. I'll merge.

Comment (michaelcypher) [inline app.py:18]: I thought of similar ideas (swapping out ZeroMQ for pure sockets / just getting the OpenCV to write it's output to the other process) but we really want to encapsulate it in one main process (as in app.py). 

There are 2 issues I can currently think of (and I can imagine a lot more) with letting the OpenCV server write or send stuff directly to ffmpeg. Firstly, we need information from the OpenCV server to start the ffmpeg process (width and height) so we either have to get the OpenCV server to send to 2 queues (app.py to start ffmpeg and raw data after to ffmpeg) or OpenCV could start ffmpeg. Both options introduce a new issue, how does OpenCV server know the ffmpeg process in the first case or how does the main server (app.py) know the ffmpeg in the second?

I think at the moment, I can't think of other, nicer ways of doing this than with subprocesses or threads but if you have any ideas let me know

**Pull request merged**